src/file-events/cpp/win_fsnotifier.cpp

```diff
@@ -182,25 +183,18 @@ void Server::startWatching(JNIEnv* env, const u16string& path) {
         return;
     }
 
-    WatchPoint* watchPoint = new WatchPoint(this, path, directoryHandle);
-
     HANDLE threadHandle = watcherThread.native_handle();
-    int ret = watchPoint->awaitListeningStarted(threadHandle);
-    switch (ret) {
-        case WATCH_LISTENING:
-            watchPoints.push_back(watchPoint);
-            break;
-        default:
-            log_severe(env, "Couldn't start listening to '%ls': %d", pathW.c_str(), ret);
-            delete watchPoint;
-            // TODO Error handling
-            break;
-    }
+    watchPoints.emplace(piecewise_construct,
```

Comment:
Should we also check here if the location is already watched?

Reply:
Good point. Fixing it.